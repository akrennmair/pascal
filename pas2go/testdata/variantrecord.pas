program test;

type foo = record
		c : integer;
		d : real;
		case bla : integer of
			1, 2, 3: ( a : real );
			3, 4, 5: ( b : string );
		end;

var x : foo;

begin
	x.c := 42;
	x.d := 23.5;
	x.bla := 1;
	x.a := 42.23;
	x.b := 'judgement day'
end.
