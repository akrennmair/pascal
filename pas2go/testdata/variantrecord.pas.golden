package main

import (
	"github.com/akrennmair/pascal/pas2go/system"
)

var _ = system.Write

// program test
func main() {
	type (
		foo struct {
			c   int
			d   float64
			bla integer `pas2go:"tagfield"`
			a   float64 `pas2go:"caselabels,1,2,3"`
			b   string  `pas2go:"caselabels,3,4,5"`
		}
	)

	var (
		x foo
	)
	_ = x

	x.c = 42
	x.d = 23.5e0
	x.bla = 1
	x.a = 42.23e0
	x.b = "judgement day"
}
